web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -1432,19 +1442,41 @@
             return defer.promise;
           },
 
+          /**
+           * @ngdoc method
+           * @methodOf gn_map.addEsriRestFromScratch:gnMap
+           * @name gnMap#addEsriRestFromScratch
+           *
+           * @description
+           * Here is the method to use when you want to add a ESRIREst layer from
+           * a url and a name (layer identifier).
+           *
+           * Return a promise with ol.Layer as data is succeed, and url/name
+           * if failure.
+           * If createOnly, we don't add the layer to the map.
+           * If the md object is given, we add it to the layer, or we try
+           * to retrieve it in the catalog
+           *
+           * @param {ol.Map} map to add the layer
+           * @param {string} url of the service
+           * @param {string} name of the layer (identifier)
```

Comment:
Although this was admittedly not clear at all before, the `name` parameter is optional I think. If not present, the name will be inferred from the URL. Why this is like that, I'm not sure... The logic around `name` and `layer` in this method is unclear to me.

Reply:
For me was quite confusing the usage of that parameters, just tried to document it. Please check as at least part of the original code was added by you, maybe you can remember the logic for that variables.

Reply:
Ok so I looked in more details and the `name` parameter is basically a stringified number used to identify a layer in a map. So `0`, `1`, `2` etc. The weird thing is that this variable is reused later on to store an arbitrary "layer name" in the form of `mapId layerNumber`, so for example `my-map 0`. On the other hand, the `layer` variable is used to store the layer name instead...

The confusing part is that a service url _can_ contain a layer name (e.g. `my-map/MapServer/0`) and as such we can infer the layer name _from the service url in case the provided layer name argument is incorrect_ (i.e. not a valid number in string format). This is all unnecessarily complicated, but I understand that this was out of your PR scope. I guess the bare minimum would be to make sure that the function is properly documented (especially the `name` and `url` arguments).